### ISSUE TITLE
Include aliases in query plan

### ISSUE THREAD

**Issue by crissi** [webonyx/graphql-php#1072]
Running lookAhead with queryplan with this query: 
```   
$resolveInfo->lookAhead(['group-implementor-fields'])->queryPlan();
```
```   
query  {
  caseById(id: 7317) {
    default: schemesCount
    closed: schemesCount(status: CLOSED)
    approved: schemesCount(status: APPROVED)
  }
}
```
Gives us:
```
{
  "fields": {
    "schemesCount": {
      "type": "Int",
      "fields": [],
      "args": {
        "status": "APPROVED"
      }
    }
  }
}

```
I suspect all fields should be included somehow in the result!

**Comment by spawnia** [comment]
I think the purpose of the query plan is to allow the server to make decisions on how to resolve the query efficiently. Given aliases are given by the client and transparently handled by the execution engine of this library, they do not seem useful to include.

**Comment by crissi** [comment]
yes, but since each alias can have different arguments like in my example. 

Here it just returns the argument of the last **schemesCount** property.

**Comment by spawnia** [comment]
Right, that is misleading. `args` should really be a list with multiple args passed to the different aliases - same with `fields` actually. Perhaps a better structure would be:

```json
{
  "fields": {
    "schemesCount": {
      "type": "Int",
      "aliases": [
        {
          "fields": [],
          "args": {}
        },
        {
          "fields": [],
          "args": {
            "status": "APPROVED"
          }
        },
        {
          "fields": [],
          "args": {
            "status": "CLOSED"
          }
        },
      ]
    }
  }
}
```

**Comment by crissi** [comment]
maybe include the alias name in there  for good measure

**Comment by jeroenschieven** [comment]
We are having the same issue. We use the queryPlan to dynamically build an ElasticSearch query. Some fields have quite some possible arguments, therefore we want to be able to alias those fields.

Unfortunately, the only alternative for now is duplicating these fields.

**Comment by zephyx** [comment]
I needed this too, so I have modified the lib to get this feature.
I don't know if it will be merged or no, but if needed for your application: https://github.com/webonyx/graphql-php/pull/1648

I use it as follow:

=> In My Query Resolver : 

$aliasArgs = $info->lookAhead()->aliasArgs();
//I can then create multiple results for each of the aliases like this:
$res[$aliasName] = $myResult;

=> In My Type Resolver : 
//I have access to all my results with $root and I have just to access to them with a 
return $root[$info->fieldNodes[0]->alias->value];

I did it this way because it was the quickest and easiest way to produce this feature, so don't hesitate if you have a better idea!

**Comment by zephyx** [comment]
Ofc, this code is an example and is not enough, you have to manage the no alias case :)

**Comment by zephyx** [comment]
Finally, I have modified the feature, and now it can be used with a method of ResolveInfo : getFieldSelectionWithAliases()

Basic example usage for a level 1 field:

- In Query Resolver:
```
public function resolve($root, $args, $context, ResolveInfo $info)
{
    $fields = $info->getFieldSelectionWithAliases();
    
    if (isset($fields['myFieldWithAliases'])) {
        foreach ($fields['myFieldWithAliases'] as $aliasName => $selectionSet) {
            $category = $selectionSet['args']['category'];
            $limit = $selectionSet['args']['limit'];
            
            //build my SQL query with args
            $q = ....
            
            $res[$aliasName] = $q->all();
        }
    }

    return $res;
}
```
    
- in Field resolver:
```
public static function resolveMyFieldWithAliasesField($root, array $args, $context, $info)
{
     if (isset($info->fieldNodes[0]->alias)) {
         return $root[$info->fieldNodes[0]->alias->value];
     }

     return $root['myFieldWithAliases'];
}
```

It has been merged with tag 15.19.0
